### PR TITLE
feat(provider/kubernetes): v2 server group manager

### PR DESCRIPTION
**Pull request by lwander**
spinnaker/spinnaker#1902

### CODE REVIEW

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
+  @PostFilter("hasPermission(filterObject.account, 'ACCOUNT', 'READ')" )
+  @RequestMapping(method = RequestMethod.GET)
+  Set<ServerGroupManager> getForApplication(@PathVariable String application) {
+    return serverGroupManagerProviders.stream()
+        .map(provider -> {
+          try {
+            return requestQueue.execute(application, () -> provider.getServerGroupManagersByApplication(application));
+          } catch (Throwable t) {
+            log.warn("Failed to read manifest " , t);
```
Comment (danielpeach) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java]: This feels kubernetes-specific.

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  @Autowired
+  public ServerGroupManagerController(List<ServerGroupManagerProvider> serverGroupManagerProviders, RequestQueue requestQueue) {
+    this.serverGroupManagerProviders = serverGroupManagerProviders;
+    this.requestQueue = requestQueue;
+  }
+
+  @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ')")
+  @PostFilter("hasPermission(filterObject.account, 'ACCOUNT', 'READ')" )
+  @RequestMapping(method = RequestMethod.GET)
+  Set<ServerGroupManager> getForApplication(@PathVariable String application) {
```
Comment (danielpeach) [inline clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java]: A server group manager's server groups might not all live in the same application, right? I wonder how the UI should show that...

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java (hunk 1 above)
Comment (lwander) [reply clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java]: Ah this was copied out of the "ManifestController", where that warning actually makes sense

File: clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java (hunk 2 above)
Comment (lwander) [reply clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java]: That's fine, if you're on a certain "cluster" tab you would only see the server groups in that app. Clicking on the server group manager could show you the list of owned server groups (even ones in different apps) in the details panel the same way we do for load balancers.
ygauhxgpjwl

Comment (lwander) [reply clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java]: Ah are you worried about server groups who's manager isn't in the current app? It's probably safe to ignore those (at least for now).

Comment (lwander) [reply clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupManagerController.java]: We could also annotate server groups with their managers, and if one points to a manager not in this app, we can add a little button like the load balancer button, but this feels like an unlikely use-case.

**Pull request merged**